src/private/MDMPerformerGroup.m

```diff
@@ -76,30 +78,64 @@ - (void)terminateIsActiveToken:(id<MDMIsActiveTokenable>)token
 #pragma mark - Private
 
 - (void)addPlan:(id<MDMPlan>)plan trace:(MDMTrace *)trace log:(MDMTransactionLog *)log {
-  [trace.committedPlans addObject:plan];
-  
+  // all named addPlan plans must first be removed before being added
+  if ([self isNamedTransactionLog:log] && !log.isRemoval) {
+    [self addPlan:plan trace:trace log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:log.target name:log.name removal:TRUE]];
+  }
+  // see if we can get a performer
+  id<MDMPerforming> performer = [self performerForPlan:plan trace:trace log:log];
+  if (performer != nil) {
+    if (plan != nil) {
+      [trace.committedPlans addObject:plan];
+    } else {
+      // this is the case whereby we are calling removePlan:named, but don't have a MDMPlan
```

Comment:
OK - will take a look at that code review.

Need to figure out exactly what data structure would be in the `removedPlans` collection as we don't always have a `MDMPlan` object when calling `removePlan:named:from`. I'm guessing `removedPlans` would contain a structure of names associated with targets.

or

Whenever we get a `removePlan:named:from` committed to the scheduler, we should look back through `committedPlans` and see if we can find a `MDMNamedPlan` which was previously added which matches the name and target. If we find a plan, copy it over into `removedPlans`